x-pack/plugins/security/server/routes/role_mapping/get.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { schema } from '@kbn/config-schema';
+import { RoleMapping } from '../../../../../legacy/plugins/security/common/model';
+import { createLicensedRouteHandler } from '../licensed_route_handler';
+import { RouteDefinitionParams } from '..';
+
+interface RoleMappingsResponse {
+  [roleMappingName: string]: Omit<RoleMapping, 'name'>;
+}
+
+export function defineRoleMappingGetRoutes(params: RouteDefinitionParams) {
+  const { clusterClient, logger, router } = params;
+
+  router.get(
+    {
+      path: '/internal/security/role_mapping/{name?}',
+      validate: {
+        params: schema.object({
+          name: schema.maybe(schema.string()),
+        }),
+      },
+    },
+    createLicensedRouteHandler(async (context, request, response) => {
+      const roleMappingsResponse: RoleMappingsResponse = await clusterClient
+        .asScoped(request)
+        .callAsCurrentUser('shield.getRoleMappings', {
+          name: request.params.name,
+        });
+
+      const expectSingleEntity = typeof request.params.name === 'string';
+
+      const mappings = Object.entries(roleMappingsResponse).map(([name, mapping]) => {
+        return {
+          name,
+          ...mapping,
+          role_templates: (mapping.role_templates || []).map(entry => {
+            return {
+              ...entry,
+              template: tryParseRoleTemplate(entry.template as string),
+            };
+          }),
+        } as RoleMapping;
+      });
+
+      if (expectSingleEntity) {
+        if (mappings.length === 0) {
+          return response.notFound();
+        }
```

Comment:
I can't seem to trigger this code path (e.g., by manually creating a URL to attempt to edit an unknown role mapping). I always get a 500 error. I tinkered with this a bit, I believe you need to wrap the clusterClient call in a try/catch and check the status code to return a 404 error if necessary.

Also could use a test case for this scenario.

Reply:
Good catch, I'll revise this!